mayavi/plugins/envisage_engine.py

```diff
@@ -118,19 +117,19 @@ def _scene_editors_changed(self, list_event):
         for scene in list_event.added:
             self.add_scene(scene)
 
-    @on_trait_change('window:opened')
+    @on_trait_change('application:window_opened')
     def _on_window_opened(self, obj, trait_name, old, new):
         """We start the engine when the window is opened."""
-        if trait_name == 'opened':
+        if trait_name == 'window_opened':
             self.start()
 
-    @on_trait_change('window:closed')
+    @on_trait_change('application:window_closed')
```

Comment:
This will fire when _any_ window in the Tasks application closes, not just the currently used window, so you could end up in a situation where the window using this engine is still active, but the engine is not.